orangecontrib/text/widgets/owstatistics.py

```diff
@@ -0,0 +1,707 @@
+import re
+from copy import copy
+from typing import Callable, List, Optional, Tuple
+
+import numpy as np
+from AnyQt.QtCore import QSize
+from AnyQt.QtWidgets import QComboBox, QGridLayout, QLabel, QLineEdit
+
+from Orange.widgets import gui
+from Orange.widgets.settings import ContextSetting
+from Orange.widgets.utils.concurrent import ConcurrentWidgetMixin, TaskState
+from Orange.widgets.utils.widgetpreview import WidgetPreview
+from Orange.widgets.widget import Input, Output, OWWidget
+from orangewidget.widget import Msg
+
+from orangecontrib.text import Corpus
+
+# those functions are implemented here since they are used in more statistics
+from orangecontrib.text.preprocess import (
+    LowercaseTransformer,
+    Preprocessor,
+    RegexpTokenizer,
+    UrlRemover,
+)
+from orangecontrib.text.widgets.utils import format_summary_details
+from orangecontrib.text.widgets.utils.context import (
+    AlmostPerfectContextHandler,
+)
+
+punctuations = [
+    ".",
```

Comment:
Probably it would be nicer to have this on a single line. Or even better, you could use
```
import string
string.punctuation
```

Reply:
I used black for code formating this is why this array is like this. Anyway, string.punction is a better option. Thanks.